File: backend/ssh_client.py
#!/usr/bin/env python3
import sys
import paramiko
import json
import time

def ssh_connect(host, username, password, command):
    """Connect to SSH and execute command"""
    try:
        # Create SSH client
        client = paramiko.SSHClient()
        client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
        
        # Connect
        client.connect(
            hostname=host,
            username=username,
            password=password,
            timeout=10,
            look_for_keys=False,
            allow_agent=False
        )
        
        # Execute command
        stdin, stdout, stderr = client.exec_command(command)
        
        # Get output
        output = stdout.read().decode('utf-8')
        error = stderr.read().decode('utf-8')
        
        # Close connection
        client.close()
        
        if error:
            return {"success": False, "error": error, "output": output}
        else:
            return {"success": True, "output": output, "error": None}
            
    except Exception as e:
        return {"success": False, "error": str(e), "output": None}

def main():
    if len(sys.argv) < 5:
        print(json.dumps({
            "success": False,
            "error": "Usage: python ssh_client.py <host> <username> <password> <command>",
            "output": None
        }))
        sys.exit(1)
    
    host = sys.argv[1]
    username = sys.argv[2]
    password = sys.argv[3]
    command = sys.argv[4]
    
    result = ssh_connect(host, username, password, command)
    print(json.dumps(result))

if __name__ == "__main__":
    main()

File: backend/ssh-client.go
package main

import (
	"fmt"
	"log"
	"os"
	"strings"
	"time"

	"golang.org/x/crypto/ssh"
)

type SSHClient struct {
	Host     string
	Username string
	Password string
	Client   *ssh.Client
}

func NewSSHClient(host, username, password string) *SSHClient {
	return &SSHClient{
		Host:     host,
		Username: username,
		Password: password,
	}
}

func (s *SSHClient) Connect() error {
	config := &ssh.ClientConfig{
		User: s.Username,
		Auth: []ssh.AuthMethod{
			ssh.Password(s.Password),
		},
		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
		Timeout:         10 * time.Second,
	}

	client, err := ssh.Dial("tcp", s.Host+":22", config)
	if err != nil {
		return fmt.Errorf("failed to connect: %v", err)
	}

	s.Client = client
	return nil
}

func (s *SSHClient) ExecuteCommand(command string) (string, error) {
	if s.Client == nil {
		return "", fmt.Errorf("not connected")
	}

	session, err := s.Client.NewSession()
	if err != nil {
		return "", fmt.Errorf("failed to create session: %v", err)
	}
	defer session.Close()

	output, err := session.CombinedOutput(command)
	if err != nil {
		return "", fmt.Errorf("command failed: %v", err)
	}

	return string(output), nil
}

func (s *SSHClient) Close() {
	if s.Client != nil {
		s.Client.Close()
	}
}

func main() {
	if len(os.Args) < 4 {
		fmt.Println("Usage: go run ssh-client.go <host> <username> <password> <command>")
		os.Exit(1)
	}

	host := os.Args[1]
	username := os.Args[2]
	password := os.Args[3]
	command := os.Args[4]

	client := NewSSHClient(host, username, password)
	
	// Connect
	if err := client.Connect(); err != nil {
		log.Fatalf("Connection failed: %v", err)
	}
	defer client.Close()

	// Execute command
	output, err := client.ExecuteCommand(command)
	if err != nil {
		log.Fatalf("Command execution failed: %v", err)
	}

	// Print output
	fmt.Print(output)
}
